ext/ffi/lib.rs

```diff
@@ -409,6 +444,18 @@ impl NativeValue {
           v8::BigInt::new_from_u64(scope, self.pointer as u64).into();
         local_value.into()
       }
+      NativeType::Struct(_) => {
+        let store = v8::ArrayBuffer::new_backing_store_from_vec(
+          self.structure_value.to_vec(),
```

Comment:
So here I think a memory leak happens. `self.structure_value` gets cloned to a new vector which is used for the `ArrayBuffer` and its lifetime is then controlled by V8. The original copy, however, is left alone and leaks.

I think the appropriate API would be `.into_inner()` from the `ManuallyDrop` type.

However, as said I think this should actually be a boxed slice. `ArrayBuffer` has a creation API for that as well.